contracts/IdleCDO.sol

```diff
@@ -941,6 +941,37 @@ contract IdleCDO is PausableUpgradeable, GuardedLaunchUpgradable, IdleCDOStorage
     revertIfTooLow = true;
   }
 
+  /// @notice allow redeems of remaining funds, if any, after default/losses
+  /// @dev can be called by the owner only
+  function setRedemptionRates() external {
+    _checkOnlyOwner();
+
+    // liquidate all ibTokens
+    uint256 _amount = IERC20Detailed(strategyToken).balanceOf(address(this)) * _strategyPrice() / ONE_TRANCHE_TOKEN;
+    _liquidate(_amount, false);
+
+    address _AATranche = AATranche;
+    uint256 _AASupply = IERC20(_AATranche).totalSupply();
+    uint256 _lastAAPrice = _tranchePrice(_AATranche);
+    // Get all underlyings in the contract
+    uint256 _underlyingBalance = IERC20(token).balanceOf(address(this));
+    // Set redeem price as if everything is going to AA 
+    uint256 redeemPrice = _underlyingBalance * ONE_TRANCHE_TOKEN / _AASupply;
+    // Check if rate is too high compared to the last saved price
+    if (redeemPrice > _lastAAPrice) {
```

Comment:
memo: in this scenario, AA holder would receive underlying proportional to lastAAPrice.